src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/HttpClientJsonExtensions.Get.AsyncEnumerable.cs

```diff
@@ -135,12 +135,29 @@ public static partial class HttpClientJsonExtensions
             JsonSerializerOptions? options,
             CancellationToken cancellationToken)
         {
-            options ??= JsonSerializerOptions.Default;
-            options.MakeReadOnly();
+            if (client is null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
 
-            var jsonTypeInfo = (JsonTypeInfo<TValue>)options.GetTypeInfo(typeof(TValue));
```

Comment:
The crux of the fix should be replacing the above with a call to the `JsonHelpers.GetJsonTypeInfo` method. See the `HttpContent` implementation for pointers.

Reply:
updated